### DISCUSSION TITLE
Is it safe to call methods like Player#setSomething() in threads beyond Ms-TickingThread without Acquirable,lock or synchronization?

### DISCUSSION THREAD

**Discussion by Anzide** [Minestom/Minestom#1277] [Help]
I guess no,but i want to get a confirmation however,
because there is no such a concept of "main thread" in Minestom,and Minestom is designed to run on multiple threads.
So here is my questions:
1.Is it safe to call methods like Player#setSkin() in threads beyond Ms-TickingThread without Acquirable,lock or synchronization?
2.If answer to 1 is no,the solution should be synchonize the Player object wherever I used it,or just call SchedulerManager.scheduleNextTick?Because there is no concept of "main thread" and Minestom is "designed for multi threads",I am not sure which thread SHOULD i run the task on.

**Comment by KrystilizeNevaDies** [comment]
If you are running the code in arbitrary threads, you should use the acquirable API. Events however are fine.
